modules/server/src/main/scala/seqexec/server/SequenceGen.scala

```diff
@@ -75,12 +75,22 @@ object SequenceGen {
   }
 
   final case class StepActionsGen[F[_]](
-    configs: Map[Resource, SystemOverrides => Action[F]],
-    post:    (HeaderExtraData, SystemOverrides) => List[ParallelActions[F]]
+    configs:          Map[Resource, SystemOverrides => Action[F]],
+    post:             (HeaderExtraData, SystemOverrides) => List[ParallelActions[F]],
+    endObserveAction: Option[SystemOverrides => Action[F]]
   ) {
-    def generate(ctx: HeaderExtraData, overrides: SystemOverrides): List[ParallelActions[F]] =
+
+    def generate(
+      ctx:       HeaderExtraData,
+      overrides: SystemOverrides
+    ): List[ParallelActions[F]] = {
+      val postActions =
+        this.endObserveAction.fold(post(ctx, overrides))(a =>
+          post(ctx, overrides) :+ NonEmptyList.one(a(overrides))
+        )
```

Comment:
Isn't a bit risky to put `endObserveAction` together with `post` ? They will run in parallel, but what if something in `post` fails?
I think it should be run after `post`.